src/impl/results_notifier.cpp

```diff
@@ -149,20 +158,31 @@ void ResultsNotifier::calculate_changes()
         m_changes = CollectionChangeBuilder::calculate(m_previous_rows, next_rows,
                                                        get_modification_checker(*m_info, *m_query->get_table()),
                                                        move_candidates);
-
         m_previous_rows = std::move(next_rows);
+
+#if REALM_ENABLE_SYNC
+        if (m_partial_sync_enabled)
+            m_previous_partial_sync_status_code = partial_sync::create_or_update_subscription(*m_sg, m_changes, *m_query, m_previous_partial_sync_status_code);
+#endif
+
     }
     else {
         m_previous_rows.resize(m_tv.size());
         for (size_t i = 0; i < m_tv.size(); ++i)
             m_previous_rows[i] = m_tv[i].get_index();
+
+#if REALM_ENABLE_SYNC
+        if (m_partial_sync_enabled)
+            m_previous_partial_sync_status_code = partial_sync::create_or_update_subscription(*m_sg, m_changes, *m_query, m_previous_partial_sync_status_code);
+#endif
     }
 }
 
 void ResultsNotifier::run()
 {
     // Table's been deleted, so report all rows as deleted
     if (!m_query->get_table()->is_attached()) {
+        // FIXME: How to handle partial sync here? Question: Why is this before need_to_run()?
```

Comment:
This is at the beginning so that all of the following code can assume that the Table still exists.